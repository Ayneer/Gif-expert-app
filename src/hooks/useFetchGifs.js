import { useEffect, useState } from 'react';
import { getGifs } from '../services/getGif';

export const useFetchGifs = (category) => {

    const [state, setstate] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        getGifs(category)
            .then(data => {
                setstate({
                    data,
                    loading: false
                })
            });
    }, [category]);

    return state;
}