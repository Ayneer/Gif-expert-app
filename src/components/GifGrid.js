import React from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

    const { loading, data } = useFetchGifs(category);


    return (
        <>
            <h3 className='animate__animated animate__zoomIn'>{category}</h3>
            {loading && 'Loading...'}
            <div className='card-grid'>
                {data.map(gif => (
                    <GifGridItem
                        key={gif.id}
                        {...gif}
                    />
                ))}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
