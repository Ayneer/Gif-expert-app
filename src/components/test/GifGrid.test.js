import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import { GifGrid } from '../GifGrid';

jest.mock('../../hooks/useFetchGifs');

describe('Gif Grid Test Suit', () => {
    test('should show the component successfully', () => {
        useFetchGifs.mockReturnValue({ loading: true, data: [] });
        const category = 'Halo';

        const wrapper = shallow(<GifGrid category={category} />);

        const htmlP = wrapper.find('p').text();
        const htmlH3 = wrapper.find('h3').text();

        expect(wrapper).toMatchSnapshot();
        expect(htmlP).toBe('Loading...');
        expect(htmlH3).toBe('Halo');
    });

    test('should show some gifs', () => {
        const data = [
            {
                id: '1',
                title: 'title mock',
                url: 'mock'
            }
        ];
        useFetchGifs.mockReturnValue({ loading: false, data });
        const category = 'Halo';

        const wrapper = shallow(<GifGrid category={category} />);

        const htmlH3 = wrapper.find('h3').text();
        const gifGridItem = wrapper.find('GifGridItem');
        const htmlGifGridItemUrl = gifGridItem.prop('url');
        const htmlGifGridItemTitle = gifGridItem.prop('title');
        const htmlGifGridItemId = gifGridItem.prop('id');

        expect(htmlH3).toBe('Halo');
        expect(htmlGifGridItemUrl).toBe('mock');
        expect(htmlGifGridItemTitle).toBe('title mock');
        expect(htmlGifGridItemId).toBe('1');
        expect(wrapper.find('p').exists()).toBeFalsy();
        expect(gifGridItem.length).toBe(data.length);
    });
});
