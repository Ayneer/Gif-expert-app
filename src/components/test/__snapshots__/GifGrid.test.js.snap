// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif Grid Test Suit should show the component successfully 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__zoomIn"
  >
    Halo
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
